$NetBSD: patch-af,v 1.3 2010/11/18 13:56:05 wiz Exp $

PR 43815:
net/avahi fails on my Linux system:

   ...
     CC     avahi_daemon-sd-daemon.o
   sd-daemon.c: In function 'sd_notify':
   sd-daemon.c:356: error: 'SOCK_CLOEXEC' undeclared (first use in this function)
   sd-daemon.c:356: error: (Each undeclared identifier is reported only once
   sd-daemon.c:356: error: for each function it appears in.)
   gmake[2]: *** [avahi_daemon-sd-daemon.o] Error 1

This is because my glibc and Kernel is not modern enough for this
version of avahi.

http://avahi.org/ticket/316

--- avahi-daemon/sd-daemon.c.orig	2010-07-13 00:45:20.000000000 +0000
+++ avahi-daemon/sd-daemon.c
@@ -41,6 +41,10 @@
 #include <stdarg.h>
 #include <stdio.h>
 
+#if !defined(SOCK_CLOEXEC) && !defined(DISABLE_SYSTEMD)
+#define DISABLE_SYSTEMD 1
+#endif
+
 #include "sd-daemon.h"
 
 int sd_listen_fds(int unset_environment) {
